FSICoupler.py

```diff
@@ -2283,11 +2321,199 @@ def setOmega(self):
+                    (solidInterfaceDisplacement_tilde1_X_Gat, solidInterfaceDisplacement_tilde1_Y_Gat, solidInterfaceDisplacement_tilde1_Z_Gat) = mpiGatherInterfaceData(solidInterfaceDisplacement_tilde1, ns, self.mpiComm, 0)
+                    
+                    if self.myid == 0:
+                        if self.FSIIter > 0: # Either information from previous time steps is re-used or not, Vk and Wk matrices are enriched only starting from the second iteration of every FSI loop
+                            delta_res = np.concatenate([res_X_Gat - solidInterfaceResidual0_X_Gat, res_Y_Gat - solidInterfaceResidual0_Y_Gat, res_Z_Gat - solidInterfaceResidual0_Z_Gat], axis=0)
+                            delta_d = np.concatenate([solidInterfaceDisplacement_tilde_X_Gat - solidInterfaceDisplacement_tilde1_X_Gat, solidInterfaceDisplacement_tilde_Y_Gat - solidInterfaceDisplacement_tilde1_Y_Gat, solidInterfaceDisplacement_tilde_Z_Gat - solidInterfaceDisplacement_tilde1_Z_Gat], axis = 0)
+                        
+                            Vk.insert(0, delta_res)
+                            Wk.insert(0, delta_d)
+                        
+                            nIt+=1
+                    
+                        Vk_mat = np.vstack(Vk).T
+                        Wk_mat = np.vstack(Wk).T
+                        
+                        Q, R = sp.linalg.qr(Vk_mat, mode='economic')
+                        
+                        s = np.dot(np.transpose(Q), -np.concatenate([res_X_Gat, res_Y_Gat, res_Z_Gat], axis=0))
+                    
+                        toll = 1e-10*np.linalg.norm(np.concatenate([res_X_Gat, res_Y_Gat, res_Z_Gat]))
+                        c = solve_upper_triangular_mod(R, s, toll)
+                    
+                        delta_ds_loc = np.split((np.dot(Wk_mat,c).T + np.concatenate([res_X_Gat, res_Y_Gat, res_Z_Gat], axis=0)),3,axis=1)
+
+                        delta_ds_loc_X = np.concatenate(delta_ds_loc[0])
+                        delta_ds_loc_Y = np.concatenate(delta_ds_loc[1])
+                        delta_ds_loc_Z = np.concatenate(delta_ds_loc[2])
+
+                        for iVertex in range(delta_ds_loc_X.shape[0]):
+                            iGlobalVertex = self.manager.getGlobalIndex('solid', self.myid, iVertex)
+                            delta_ds[iGlobalVertex] = (delta_ds_loc_X[iVertex], delta_ds_loc_Y[iVertex], delta_ds_loc_Z[iVertex])
+                    
+                    # --- Go back to parallel run --- #
+                    mpiBarrier(self.mpiComm)
+                    delta_ds.assemble()
+                    #self.interfaceInterpolator.solidInterfaceDisplacement.data_X += np.concatenate(delta_ds_loc_X[0])
+                    #self.interfaceInterpolator.solidInterfaceDisplacement.data_Y += np.concatenate(delta_ds_loc_Y[1])
+                    #self.interfaceInterpolator.solidInterfaceDisplacement.data_Z += np.concatenate(delta_ds_loc_Z[2])
```

Comment:
Those commented lines can now definitely  be removed. I'll do it before merging this in.